javascript/src/test/pages/RoomSlots/RoomSlots.test.js

```diff
@@ -0,0 +1,141 @@
+      startTime: "15:00:00",
+      endTime: "15:50:00",
+    },
+  ];
+
+  const user = {
+    name: "test user",
+  };
+
+  const getAccessTokenSilentlySpy = jest.fn();
+  const mutateSpy = jest.fn();
+
+  beforeEach(() => {
+    useAuth0.mockReturnValue({
+      admin: undefined,
+      getAccessTokenSilently: getAccessTokenSilentlySpy,
+      user: user
+    });
+
+    useSWR.mockImplementation((key, _getter) => {
+      if (key[0] === "/api/public/roomslot") {
+        return {
+          data: roomSlotList,
+          error: undefined,
+          mutate: mutateSpy,
+        };
+      } else {
+        return {
+          data: {
+            user: "test user",
+            role: "admin"
+          }
+        }
+      }
+    });
+  });
+  
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+  
+    test("renders without crashing", () => {
+      render(<RoomSlots />);
+    });
+
+    test("renders loading while room slots list is undefined", () => {
+      useSWR.mockReturnValue({
+        data: undefined,
+        error: undefined,
+        mutate: mutateSpy,
+      });
+      const { getByAltText } = render(<RoomSlots />);
+      const loading = getByAltText("Loading");
+      expect(loading).toBeInTheDocument();
+    });
+
+    test("renders an error message when there is an error", () => {
+      useSWR.mockReturnValue({
+        data: undefined,
+        error: new Error("this is an error"),
+        mutate: mutateSpy,
+      });
+      const { getByText } = render(<RoomSlots />);
+      const error = getByText(/error/);
+      expect(error).toBeInTheDocument();
+    });
+
+    test("renders new room slot button when admin", async () => {
+      const pushSpy = jest.fn();
+      useHistory.mockReturnValue({
+        push: pushSpy
+      });
+
+      const { getByText } = render(<RoomSlots />);
+      const newRoomSlotButton = getByText("New Room Slot");
+      expect(newRoomSlotButton).toBeInTheDocument();
+      userEvent.click(newRoomSlotButton);
+
+      await waitFor(() => expect(pushSpy).toHaveBeenCalledTimes(1));
+    });
+
+  //   test("can delete an office hour", async () => {
+  //     const fakeDeleteFunction = jest.fn();
+  //     buildDeleteOfficeHour.mockReturnValue(fakeDeleteFunction);
+  //     const { getAllByTestId } = render(<OfficeHours />);
+  //     const deleteButtons = getAllByTestId("delete-button");
+  //     userEvent.click(deleteButtons[0]);
+  //     await waitFor(() => expect(fakeDeleteFunction).toHaveBeenCalledTimes(1));
+  //   });
+  
+  //   test("can click to add an office hour", async () => {
+  
+  //     const pushSpy = jest.fn();
+  //     useHistory.mockReturnValue({
+  //       push: pushSpy
+  //     });
+  
+  //     const { getByText } = render(<OfficeHours />);
+  //     const newOfficeHourButton = getByText("New Office Hour");
+  //     userEvent.click(newOfficeHourButton);
+  
+  //     await waitFor(() => expect(pushSpy).toHaveBeenCalledTimes(1));
+  //   });
```

Comment:
If these tests are outdated, can you remove them? If they're commented out due to some other issues, can you use `test.skip` instead?